pkg/snmp/snmpparse/config_snmp.go

```diff
@@ -97,12 +98,15 @@ func parseConfigSnmpMain(conf config.Component) ([]SNMPConfig, error) {
 		},
 	)
 	//the UnmarshalKey stores the result in mapstructures while the snmpconfig is in yaml
-	//so for each result of the Unmarshal key we storre the result in a tmp SNMPConfig{} object
-	err := conf.UnmarshalKey("snmp_listener.configs", &configs, opt)
-	if err != nil {
-		fmt.Printf("unable to get snmp config from snmp_listener: %v", err)
-		return nil, err
+	//so for each result of the Unmarshal key we store the result in a tmp SNMPConfig{} object
+	errSnmpListenerConfig := conf.UnmarshalKey("snmp_listener.configs", &configs, opt)
+	errNetworkDevicesSnmpListenerConfig := conf.UnmarshalKey("network_devices.snmp_listener.configs", &configs, opt)
+
+	if errSnmpListenerConfig != nil && errNetworkDevicesSnmpListenerConfig != nil {
```

Comment:
Do you know why we have this code twice? I'm surprised

Reply:
it is slightly different, here we unmarshal `snmp_listener.configs` and in the other one we unmarshal `snmp_listener` 
this one is called in the `scanDevice` and in the `snmpwalk` method but I am not sure to understand the difference

Reply:
I created [a new issue](https://datadoghq.atlassian.net/browse/NDMII-2661)